tools/sigclient/pkg/query/query.go

```diff
@@ -909,3 +910,84 @@ func RunQueryFromFileAndOutputResponseTimes(dest string, filepath string, queryR
 
 	log.Infof("RunQueryFromFileAndOutputResponseTimes: Query results written to CSV file: %v", queryResultFile)
 }
+
+func RunBenchmarkUUIDQuery(uuidList []string, batchID int, wg *sync.WaitGroup, semaphore chan struct{}) {
+
+	defer wg.Done()
+	defer func() { <-semaphore }()
+	for _, uuid := range uuidList {
+
+		data := map[string]interface{}{
+			"state":         "query",
+			"searchText":    "ident=" + uuid,
+			"startEpoch":    "now-1d",
+			"endEpoch":      "now",
+			"indexName":     "*",
+			"queryLanguage": "Splunk QL",
+		}
+		evaluationType := "total"
+		relation := "gt"
+		expectedValue := "0"
+		// create websocket connection
+		conn, _, err := websocket.DefaultDialer.Dial("ws://localhost:5122/api/search/ws", nil)
+		if err != nil {
+			log.Fatalf("RunBenchmarkUUIDQuery: Error connecting to WebSocket server: %v", err)
+			return
+		}
+		defer conn.Close()
+
+		err = conn.WriteJSON(data)
+		if err != nil {
+			log.Fatalf("RunBenchmarkUUIDQuery : Received err message from server: %+v\n", err)
+		}
+
+		readEvent := make(map[string]interface{})
+		for {
+			err = conn.ReadJSON(&readEvent)
+			if websocket.IsCloseError(err, websocket.CloseNormalClosure) {
+				//WebSocket closed normally (1000).
+				break
+			} else if err != nil {
+				log.Infof("RunBenchmarkUUIDQuery : Received error from server: %+v\n", err)
+				break
+			}
+			switch readEvent["state"] {
+			case "RUNNING", "QUERY_UPDATE":
+			case "COMPLETE":
+				for eKey, eValue := range readEvent {
+					if evaluationType == "total" && eKey == "totalMatched" {
+						var hits bool
+						var finalHits float64
```

Comment:
nit: this naming is confusing to me. What does `hits` as a bool mean? Is it more aptly `hitSomething`?

Reply:
renamed it to uuidFound